pkg/controllers/workspace_controller.go

```diff
@@ -180,6 +204,81 @@ func (c *WorkspaceReconciler) deleteWorkspace(ctx context.Context, wObj *kaitov1
 	return c.garbageCollectWorkspace(ctx, wObj)
 }
 
+func (c *WorkspaceReconciler) updateControllerRevision(ctx context.Context, wObj *kaitov1alpha1.Workspace) error {
+	currentHash := computeHash(wObj)
+	annotations := wObj.GetAnnotations()
+
+	latestHash, exists := annotations[WorkspaceRevisionAnnotation]
+	if exists && latestHash == currentHash {
+		return nil
+	}
+
+	data := map[string]string{"hash": currentHash}
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("failed to marshal revision data: %w", err)
+	}
+
+	revisions := &appsv1.ControllerRevisionList{}
+	if err := c.List(ctx, revisions, client.InNamespace(wObj.Namespace), client.MatchingLabels{WorkspaceNameLabel: wObj.Name}); err != nil {
+		return fmt.Errorf("failed to list revisions: %w", err)
+	}
+	sort.Slice(revisions.Items, func(i, j int) bool {
+		return revisions.Items[i].Revision < revisions.Items[j].Revision
+	})
+
+	revisionNum := int64(1)
+
+	if exists {
+		revisionNum = revisions.Items[len(revisions.Items)-1].Revision + 1
+	}
+
+	newRevision := &appsv1.ControllerRevision{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", wObj.Name, currentHash[:8]),
+			Namespace: wObj.Namespace,
+			Labels: map[string]string{
+				WorkspaceNameLabel: wObj.Name,
+			},
+			Annotations: map[string]string{
+				WorkspaceRevisionAnnotation: currentHash,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(wObj, kaitov1alpha1.GroupVersion.WithKind("Workspace")),
+			},
+		},
+		Revision: revisionNum,
+		Data:     runtime.RawExtension{Raw: jsonData},
+	}
+
+	annotations[WorkspaceRevisionAnnotation] = currentHash
+	wObj.SetAnnotations(annotations)
+	if err := c.Update(ctx, wObj); err != nil {
+		return fmt.Errorf("failed to update Workspace annotations: %w", err)
+	}
+
+	if err := c.Create(ctx, newRevision); err != nil {
+		return fmt.Errorf("failed to create new ControllerRevision: %w", err)
+	}
+
+	const maxRevisionHistoryLimit = 10
```

Comment:
Can you move this to a global const value? https://github.com/Azure/kaito/blob/main/pkg/utils/consts/consts.go